src/go/gcsrunner/transform_config.go

```diff
@@ -101,29 +98,59 @@ func transformConfigBytes(config []byte, opts FetchConfigOptions) ([]byte, error
 
 func transformEnvoyConfig(bootstrap *bootstrappb.Bootstrap, opts FetchConfigOptions) error {
 	listeners := bootstrap.GetStaticResources().GetListeners()
-	if len(listeners) != 1 {
-		return fmt.Errorf("expected exactly 1 listener, got: %d", len(listeners))
+	if len(listeners) == 0 {
+		return fmt.Errorf("expected at least 1 listener, got: 0")
+	}
+	// Require http_listener or https_listener, as the absence of these incidates
+	// a broken config.
+	// Note that loopback_listener is not required.
+	httpListenerTransformed := false
+	for _, l := range listeners {
+		switch l.GetName() {
+		// The default listener created by this project's configgenerator package.
+		case "http_listener", "https_listener":
```

Comment:
Can you move "http_listener" and "https_listener" in listener_generator to util and make them shared?